sys-block/gpart/patches/gpart-0.3.patchset

```diff
@@ -0,0 +1,91 @@
+From 83e2443543ac0162726c44de94fcbc35fa03eb01 Mon Sep 17 00:00:00 2001
+From: localanu <localanu@gmail.com>
+Date: Thu, 4 Jan 2018 07:01:46 +0000
+Subject: Try to add support to haiku
+
+
+diff --git a/autogen.sh b/autogen.sh
+old mode 100644
+new mode 100755
+diff --git a/src/disku.c b/src/disku.c
+index d50cebf..106f2ff 100644
+--- a/src/disku.c
++++ b/src/disku.c
+@@ -33,6 +33,9 @@
+ #include <sys/disk.h>
+ #endif
+ 
++#if defined(__HAIKU__)
++#include <Drivers.h>
++
+ #include <unistd.h>
+ 
+ static void geometry_from_num_sectors(struct disk_geom *g, uint64_t nsects)
+@@ -95,8 +98,21 @@ static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
+ 	g.d_nsecs = o / u;
+ 	g.d_c = g.d_nsecs / g.d_h / g.d_s;
+ }
++
++
++#elif defined(__HAIKU__)
++static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
++{
++	device_geometry dg;
++	ioctl(d->d_fd, B_GET_GEOMETRY, & dg);
++	g -> d_h = dg.head_count;
++	g -> d_c = dg.cylinder_count;
++	g -> d_s = dg.sectors_per_track * dg.cylinder_count;
++}
++#endif
++
+ #else
+-#error Only Linux and FreeBSD supported
++#error Only Linux, FreeBSD, and Haiku supported
+ #endif
+ 
+ /*
+diff --git a/src/gm_btrfs.c b/src/gm_btrfs.c
+index cdf0a36..155bb1d 100644
+--- a/src/gm_btrfs.c
++++ b/src/gm_btrfs.c
+@@ -12,7 +12,7 @@
+  * Created:   20.11.2015 <mwilck@arcor.de>
+  *
+  */
+-
++#define _BSD_SOURCE
+ #include <string.h>
+ #include <errno.h>
+ #include <endian.h>
+diff --git a/src/gpart.h b/src/gpart.h
+index e6e213c..2e60a54 100644
+--- a/src/gpart.h
++++ b/src/gpart.h
+@@ -17,8 +17,11 @@
+ #ifndef _GPART_H
+ #define _GPART_H
+ 
++#define __BSD_SOURCE
++
+ #include "config.h"
+ 
++#include <endian.h>
+ #include "errmsgs.h"
+ #include "l64seek.h"
+ 
+diff --git a/src/l64seek.h b/src/l64seek.h
+index 5e8dc4b..cb7969a 100644
+--- a/src/l64seek.h
++++ b/src/l64seek.h
+@@ -27,7 +27,7 @@
+  * offsets.
+  */
+ 
+-typedef loff_t off64_t;
++typedef off_t off64_t;
```

Comment:
Please don't do this. For easier upstreaming, wrap this under a conditional instead

```diff
--typedef loff_t off64_t;
-+typedef off_t off64_t;
++#ifdef __HAIKU__
++typedef off_t off64_t;
++#else
++typedef loff_t off64_t;
++#endif